ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PreExecutingSubmissionValidator.scala

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator.preexecution
+
+import java.time.Instant
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlStateKey,
+  DamlStateValue,
+  DamlSubmission
+}
+import com.daml.ledger.participant.state.kvutils.KeyValueCommitting.PreExecutionResult
+import com.daml.ledger.participant.state.kvutils.api.LedgerReader
+import com.daml.ledger.participant.state.kvutils.{Bytes, Envelope, Fingerprint, KeyValueCommitting}
+import com.daml.ledger.participant.state.v1.ParticipantId
+import com.daml.ledger.validator.batch.BatchedSubmissionValidator
+import com.daml.ledger.validator.preexecution.PreExecutionCommitResult.ReadSet
+import com.daml.ledger.validator.{StateKeySerializationStrategy, ValidationFailed}
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.{Metrics, Timed}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+sealed case class PreExecutionOutput[WriteSet](
```

Comment:
I'd expect a comment here explaining the meaning of the fields.

Reply:
Will do later.